backend/src/main/java/com/neuefische/team2/backend/restaurant/RestaurantService.java

```diff
@@ -61,4 +63,29 @@ public void deleteRestaurant(String id) {
         this.findRestaurantById(id);
         restaurantRepository.deleteById(id);
     }
+
+    public Restaurant addCommentToRestaurant(String id, String commentText) throws ResourceNotFoundException {
+        logger.info("Trying to add comment to restaurant with ID {}", id);
+
+        Restaurant restaurant = this.findRestaurantById(id);
+        List<Restaurant.Comment> updatedComments = new ArrayList<>(restaurant.comments() != null ? restaurant.comments() : new ArrayList<>());
+        Restaurant.Comment comment = new Restaurant.Comment(commentText, System.currentTimeMillis());
+        updatedComments.add(comment);
+
+        Restaurant updatedRestaurant = new Restaurant(
+                restaurant.id(),
+                restaurant.title(),
+                restaurant.city(),
+                updatedComments
+        );
+        return restaurantRepository.save(updatedRestaurant);
+    }
+
+
+    public List<Restaurant.Comment> getCommentsForRestaurant(String id) throws ResourceNotFoundException {
```

Comment:
Please use a comment record.

Reply:
What is thrown here and why? And why is it a custom ResourceNotFoundException? And why is it marked as thrown even though it is an unchecked exception?